pwiz_tools/Skyline/Executables/SkylineBatch/SkylineBatch/ConfigRunner.cs

```diff
@@ -200,6 +241,92 @@ public void WriteBatchCommandsToFile(CommandWriter commandWriter, int startStep)
             }
         }
 
+        private async Task DownloadData()
+        {
+            var mainSettings = Config.MainSettings;
+            var server = mainSettings.Server;
+            if (!server.Validated)
+                server.Validate();
+
+            var allFiles = server.GetServerFiles;
+            var fileNames = allFiles.Keys;
+            var existingDataFiles = Directory.GetFiles(mainSettings.DataFolderPath).Length;
+            var dataFilter = new Regex(mainSettings.Server.DataNamingPattern);
+            var downloadingFilesEnum =
+                from name in fileNames
+                where dataFilter.IsMatch(name)
+                select name;
+            var downloadingFiles = downloadingFilesEnum.ToList();
+            if (downloadingFiles.Count == existingDataFiles && !Program.FunctionalTest) return;

```

Comment:
It looks like you're assuming that all of the files in the folder are ones that you were supposed to download (so that if the number of files in the folder match, you can assume that you're done).

Usually there are ways for bogus files to end up in there so that's not a safe assumption to make.

Reply:
Good point. It checks that the files exist and are the correct size now.